### PR TITLE
Event time analysis changes

### PR DESCRIPTION
**Pull request by topepo**
Analogous to tidymodels/tune#619

Added `eval_time` arguments to top-level tuning functions and made corresponding changes under-the-hood.

### CODE REVIEW

**Comment by topepo** [comment]
Unit tests will be in extratests repo

File: R/tune_sim_anneal.R [hunk 1]
```diff
@@ -294,6 +306,7 @@ tune_sim_anneal_workflow <-
 
     metrics <- tune::check_metrics(metrics, object)
     metrics_name <- names(attr(metrics, "metrics"))[1]
+    metrics_time <- eval_time[1]
```
Comment (hfrick) [inline R/tune_sim_anneal.R:309]: So if `eval_time` is a vector, we only ever use the first element? I'd say we should document that.

File: R/tune_race_anova.R [hunk 2]
```diff
@@ -22,6 +22,10 @@
 #' @param metrics A [yardstick::metric_set()] or `NULL`.
 #' @param control An object used to modify the tuning process. See
 #'  [control_race()] for more details.
+#' @param eval_time A numeric vector of time points where dynamic event time
+#' metrics should be computed (e.g. the time-dependent ROC curve, etc). The
+#' values should be non-negative and should probably be no greater then the
+#' largest event time in the training set.
```
Comment (hfrick) [inline R/tune_race_anova.R:28]:
```suggestion
#' values must be non-negative and should probably be no greater than the
#' largest event time in the training set.
```

How important do you consider this caution about "greater than the largest event time"? I think we should either add an explanation as to why (can be as short as "it's extrapolating") or remove it. If we want to keep it, and it gets a bit long, we can put it in the Details section. In the context of the argument definition a description with "should probably" seems too vague.

File: R/tune_sim_anneal.R (hunk 1 above)
Comment (topepo) [reply R/tune_sim_anneal.R:309]: No, that object is used to guide the optimization or racing. We compute all of them but use one for analysis.

File: R/tune_race_anova.R (hunk 2 above)
Comment (topepo) [reply R/tune_race_anova.R:28]: I _think_ that this limitation is due to KM curves, which affects the probability of censoring and other aspects. It is extrapolation, but I think that the main impediment is related to "cannot compute" more than anything. 

I agree about the vagueness.

File: R/tune_sim_anneal.R [hunk 3]
```diff
@@ -21,7 +21,11 @@
 #' [tune_race_anova()]) or a positive integer. If the initial object was a
 #' sequential search method, the simulated annealing iterations start after the
 #' last iteration of the initial results.
-#' @param control The results of `control_sim_anneal()`.
+#' @param control The results of [control_sim_anneal()].
+#' @param eval_time A numeric vector of time points where dynamic event time
+#' metrics should be computed (e.g. the time-dependent ROC curve, etc). The
+#' values should be non-negative and should probably be no greater then the
+#' largest event time in the training set.
```
Comment (hfrick) [inline R/tune_sim_anneal.R:28]:
```suggestion
#' largest event time in the training set. The first time point in `eval_time` is the
#' one that will be optimized.
```

File: R/tune_race_anova.R (hunk 2 above)
Comment (hfrick) [reply R/tune_race_anova.R:28]: I've been playing with the censoring model and yes, predictions of censoring probability for "out of range" times, i.e. times greater than the largest event time, are `NA` (but the prediction itself doesn't error).

**Pull request merged**